### PR TITLE
ddl: don't delay for async commit if MDL is enabled

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: ref #39196

Problem Summary:

No need to delay for async commit if MDL is enabled.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

File: ddl/ddl.go [hunk 1]
```diff
@@ -1329,6 +1329,12 @@ type RecoverSchemaInfo struct {
 // It should be called before any DDL that could break data consistency.
 // This provides a safe window for async commit and 1PC to commit with an old schema.
 func delayForAsyncCommit() {
+	if variable.EnableMDL.Load() {
```
Comment (sticnarf) [inline ddl/ddl.go:1332]: Is it possible to switch off MDL during an ongoing DDL and break some constraints?

Comment (wjhuang2016) [reply ddl/ddl.go:1332]: No, we forbid to switch MDL if there is any DDL running

**Review by sticnarf — APPROVED** [review]
I cannot think of possible cases of inconsistency after MDL is enabled now. I hope our tests will be still happy in the following days.

**Pull request merged**